src/datacrumbs/explorer/probe_explorer.cpp

```diff
@@ -417,11 +440,67 @@ std::vector<std::shared_ptr<Probe>> ProbeExplorer::extractProbes() {
   DC_LOG_TRACE("ProbeExplorer::extractProbes - end");
   return probes;
 }
+void ProbeExplorer::create_exclusion_file(std::vector<std::shared_ptr<Probe>> probes) {
+  DC_LOG_TRACE("ProbeExplorer::create_exclusion_file - start");
+  json_object* jexarray = json_object_new_array();
+  // Serialize each probe to JSON without functions
+  for (const auto& probe : probes) {
+    json_object* jexclude = nullptr;
+    switch (probe->type) {
+      case ProbeType::SYSCALLS:
+        jexclude = std::dynamic_pointer_cast<SysCallProbe>(probe)->toJson(false);
+        break;
+      case ProbeType::KPROBE:
+        jexclude = std::dynamic_pointer_cast<KProbe>(probe)->toJson(false);
+        break;
+      case ProbeType::UPROBE:
+        jexclude = std::dynamic_pointer_cast<UProbe>(probe)->toJson(false);
+        break;
+      case ProbeType::USDT:
+        jexclude = std::dynamic_pointer_cast<USDTProbe>(probe)->toJson(false);
+        break;
+      case ProbeType::CUSTOM:
+        jexclude = std::dynamic_pointer_cast<CustomProbe>(probe)->toJson(false);
+        break;
+      default:
+        DC_LOG_ERROR("Unknown probe type encountered.");
+        continue;  // Skip unknown types
+    }
+    if (!jexclude) {
+      DC_LOG_ERROR("Failed to serialize probe for exclusion: %s", probe->name.c_str());
+      continue;  // Skip serialization failure
+    }
+    json_object_array_add(jexarray, jexclude);
+  }
+  if(!configManager_->probe_exclusion_file_path.empty() &&
+      !std::filesystem::exists(configManager_->probe_exclusion_file_path)) {
+      const char* exclude_json_str = json_object_to_json_string_ext(jexarray, JSON_C_TO_STRING_PRETTY);
+      std::ofstream ofs(configManager_->probe_exclusion_file_path);
+      if (ofs.is_open()) {
+        ofs << exclude_json_str;
+        ofs.close();
+      } else {
+        DC_LOG_ERROR("Failed to open file: %s", configManager_->probe_exclusion_file_path.c_str());
+      }
+    }
+
+  DC_LOG_TRACE("ProbeExplorer::create_exclusion_file - end");
+}
+
+
 
 // Writes extracted probes to a JSON file and returns the probe list
 std::vector<std::shared_ptr<Probe>> ProbeExplorer::writeProbesToJson() {
   DC_LOG_TRACE("ProbeExplorer::writeProbesToJson - start");
   auto probes = extractProbes();
+  if(probes.empty()) {
+    DC_LOG_WARN("No valid probes extracted. Skipping JSON write.");
+    return probes;
+  }
+  if(!configManager_->probe_exclusion_file_path.empty() &&
+      !std::filesystem::exists(configManager_->probe_exclusion_file_path)) {
+        create_exclusion_file(probes);
+      }
```

Comment:
```suggestion
  if(!configManager_->probe_exclusion_file_path.empty() &&
      !std::filesystem::exists(configManager_->probe_exclusion_file_path)) {
        create_exclusion_file(probes);
  }
```

Fix spacing. Maybe have clang-formatter.